Tutorial updates with recent PRs + feedback from collaboration meeting (#406)
POSYDON-code/POSYDON: docs/_source/components-overview/pop_syn/binary_star.rst

The BinaryStar object
======================

The ``BinaryStar`` object is composed of two ``SingleStar`` objects and contains the current and past states of the binary.
Only parameters in the ``BINARYPROPERTIES`` list are stored in the history. 

> The current parameter value of the star object is accessed as, e.g. ``binary.orbital_period`` while its past history with ``binary.orbital_period_history``.

Reviewer: ```suggestion
The current parameter value of the star object is accessed as, e.g. ``binary.orbital_period`` while its past history is via ``binary.orbital_period_history``.
```

> The two stars are accesses as, e.g. ``binary.star_1.mass`` while his past history with ``binary.star_1.mass_history``.

Reviewer 2: accesses -> accessed

> The two stars are accesses as, e.g. ``binary.star_1.mass`` while his past history with ``binary.star_1.mass_history``.

Reviewer: ```suggestion
The two stars are accessed as, e.g (for star 1). ``binary.star_1.mass`` while its past history is via ``binary.star_1.mass_history``.
```

The binary properties are defined as follows

.. list-table:: BINARYPROPERTIES
  :header-rows: 1

>   :widths: 50 150

Reviewer 2: I'm not sure, whether it is a good idea to use fixed table widths. This might cause displaying issues on devices with small screens. Have you checked, that it displays well on e.g. a smart phone?

Author: So this works fine with different sizes and automatically readjusts

Additional scalar properties can be added during the evolution.

Since they do not change over time, they are not stored in the history.
These can requested and will be stored in the output oneline (See the :ref:`Synthetic Population<synthetic-population>` and :ref:`Population Parameter Guide<pop-params-guide>` for more information).

Additional columns
~~~~~~~~~~~~~~~~~~

The additional columns are defined as follows:

.. list-table:: Additional columns
  :header-rows: 1
  :widths: 50 150

  * - Properties
    - Descriptions
  * - ``interp_class_HMS_HMS``
    - The interpolation class for the HMS_HMS phase.
  * - ``interp_class_CO_HMS_RLO``
    - The interpolation class for the CO_HMS_RLO phase.
  * - ``interp_class_CO_HeMS``
    - The interpolation class for the CO_HeMS phase.
  * - ``interp_class_CO_HeMS_RLO``
    - The interpolation class for the CO_HeMS_RLO phase.
  * - ``mt_history_HMS_HMS``
    -  The mass transfer history for the HMS_HMS phase.
  * - ``mt_history_CO_HMS_RLO``
    - The mass transfer history for the CO_HMS_RLO phase.
  * - ``mt_history_CO_HeMS``
    - The mass transfer history for the CO_HeMS phase.
  * - ``mt_history_CO_HeMS_RLO``

>     - The mass transfer history for the CO_HeMS_RLO phase.

Reviewer 2: I'm wondering, why we don't record stuff about the detached step and why we always have a column for each binary gird.
It might be more for a future PR (put in issue): this looks like a restriction to only go to each of the binary grid phases in maximum once per binary. I think, that takes away flexibility without any need.

Author: For `mt_history`, it makes sense to not record the detached step.
For `interp_class`, it might be useful

  * - ``maxtime``
    - Max time of the evolution was reached.
  * - ``FAILED``
    - The evolution failed.

Event options
~~~~~~~~~~~~~

Binary events are defined according to the following table:

.. list-table:: Events
  :header-rows: 1
  :widths: 10 30

  * - Event
    - Description
  * - ``ZAMS``
    - Zero Age Main Sequence
  * - ``CC1``
    - Core collapse of star 1.
  * - ``CC2``
    - Core collapse of star 2.
  * - ``oRLO1``
    - The binary is at onset of Roche Lobe overflow, star 1 is overfilling the RL.
  * - ``oRLO2``
    - The binary is at onset of Roche Lobe overflow, star 2 is overfilling the RL.
  * - ``oCE1``
    - The binary is at the onset of Common Envelope initiated by star 1.
  * - ``oCE2``
    - The binary is at the onset of Common Envelope initiated by star 2.
  * - ``oDoubleCE1``
    - The binary is at the onset of Double Common Envelope initiated by star 1. 
      Both stars are post main-sequence.
  * - ``oDoubleCE2``
    - The binary is at the onset of Double Common Envelope initiated by star 2. 
      Both stars are post main-sequence.
  * - ``CO_contact``
    - The binary reached contact in the compact object phase.

>   * - ``redirect_from_ZAMS``

Reviewer 3: Just making a note that we should probably add more info about the redirect steps here once we merge PR #421, or we should add changes to the docs in that PR if we accept this first.

Author: I would suggest accepting this PR first and making the changes in PR421 or in a separate PR 👍
